irohad/model/model_crypto_provider.hpp

```diff
@@ -58,6 +58,20 @@ namespace iroha {
        * @return signed block
        */
       virtual Block sign(const Block &block) const = 0;
+
+      /**
+       * Method for signing a transaction with stored keypair
+       * @param transaction - transaction for signing
+       * @return signed transaction
+       */
+      virtual Transaction sign(const Transaction &transaction) const = 0;
+
+      /**
+       * Method for signing a query with stored keypair
+       * @param query - query to sign
+       * @return signed query
+       */
+      virtual std::shared_ptr<const Query> sign(const Query &query) const = 0;
```

Comment:
No point to check query, just assume that it's fine. You don't check that in tx or block, do you?